dsgrid/dataset/dataset_schema_handler_standard.py

```diff
@@ -40,10 +54,121 @@ def get_unique_dimension_rows(self):
         """Get distinct combinations of remapped dimensions, including id.
         Check each col in combination for null value."""
         dim_table = self._remap_dimension_columns(self._load_data_lookup).distinct()
-        self._check_null_value_in_unique_dimension_rows(dim_table)
-
+        check_null_value_in_unique_dimension_rows(dim_table)
         return dim_table
 
+    def make_project_dataframe(self):
+        # TODO: Can we remove NULLs at registration time?
+        lk_df = self._load_data_lookup.filter("id is not NULL")
+        lk_df = self._remap_dimension_columns(lk_df)
+        ld_df = self._remap_dimension_columns(
+            self._load_data,
+            # Some pivot columns may have been removed.
+            pivoted_columns=set(self._load_data.columns).intersection(
+                self.get_pivoted_dimension_columns()
+            ),
+        )
+        # TODO: handle fraction application
+        # Currently this requires fraction = 1.0
+        ld_df = ld_df.join(lk_df, on="id").drop("id")
+        # Map the time here because some columns may have been collapsed above.
+        ld_df = self._convert_time_dimension(ld_df)
+        return ld_df
+
+    def make_project_dataframe_from_query(self, context: QueryContext, project_config):
+        ld_df = self._load_data
+        lk_df = self._load_data_lookup.filter("id is not NULL")
+
+        self._check_aggregations(context)
+        lk_df, ld_df = self._prefilter_dataset(context, lk_df, ld_df)
+
+        lk_df = self._remap_dimension_columns(lk_df)
+        # Some pivoted columns may have been removed in pre-filtering.
+        pivoted_columns = set(ld_df.columns).intersection(self.get_pivoted_dimension_columns())
+        ld_df = self._remap_dimension_columns(ld_df, pivoted_columns=pivoted_columns)
+
+        pivoted_columns = set(ld_df.columns).intersection(
```

Comment:
I think you can just use `self.get_pivoted_dimension_columns_mapped_to_project()` here.

Reply:
Agreed. I can assert that there is a full match.

Reply:
I think `self.get_pivoted_dimension_columns_mapped_to_project()` already does the intersection stuff

Reply:
```
pivoted_columns = self.get_pivoted_dimension_columns_mapped_to_project()
assert not self.get_pivoted_dimension_columns_mapped_to_project().difference(ld_df.columns)
```